### PR TITLE
Add environment variables support for Client configuration

### PR DESCRIPTION
**Pull request by clamoriniere**
Support of 3 environment variables for Client configuration:
- "DD_AGENT_HOST" used to provide the Agent hostname.
- "DD_DOGSTATSD_PORT" used to provide the DogStatsD port.
- "DD_ENTITY_ID" used to provide an identifier to the Client.

### CODE REVIEW

File: src/StatsdClient/Statsd.cs [hunk 1]
```diff
@@ -179,11 +207,32 @@ private static string EscapeContent(string content)
                 .Replace("\n", "\\n");
         }
 
+        private static string ConcatTags(ref string[] constantTags,ref string[] tags)
```
[the remark this answers is not in the document]
Comment (clamoriniere) [reply src/StatsdClient/Statsd.cs]: Ok, thank you for the info

File: src/StatsdClient/StatsdUDP.cs [hunk 2]
```diff
@@ -15,19 +15,54 @@ public class StatsdUDP : IDisposable, IStatsdUDP
...
 
+        public StatsdUDP(int maxUDPPacketSize = StatsdConfig.DefaultStatsdMaxUDPPacketSize)
+        : this(GetHostNameFromEnvVar(),GetPortFromEnvVar(StatsdConfig.DefaultStatsdPort),maxUDPPacketSize)
+        {
+        }
         public StatsdUDP(string name, int port, int maxUDPPacketSize = StatsdConfig.DefaultStatsdMaxUDPPacketSize)
         {
-            Name = name;
             Port = port;
+            if (Port == -1)
```
[the remark this answers is not in the document]
Comment (clamoriniere) [reply src/StatsdClient/StatsdUDP.cs]: ok

File: src/StatsdClient/Statsd.cs [hunk 3]
```diff
@@ -192,14 +241,32 @@ public class Distribution : Metric { }
...
+            if (!string.IsNullOrEmpty(entityID))
+            {
+                string entityIDTag = string.Format("{0}:{1}", ENTITY_ID_INTERNAL_TAG_KEY, entityID);
+                if (constantTags == null)
+                {
+                    _constantTags = new string[]{entityIDTag};
+                }
+                else
+                {
+                    Array.Resize(ref _constantTags, _constantTags.Length + 1);
```
[the remark this answers is not in the document]
Comment (clamoriniere) [reply src/StatsdClient/Statsd.cs]: done

File: src/StatsdClient/Statsd.cs [hunk 4]
```diff
@@ -179,11 +207,32 @@ private static string EscapeContent(string content)
                 .Replace("\n", "\\n");
         }
 
+        private static string ConcatTags(ref string[] constantTags,ref string[] tags)
+        {
```
[the remark this answers is not in the document]
Comment (clamoriniere) [reply src/StatsdClient/Statsd.cs]: I took your remarks, and update the PR.

**Comment by clamoriniere** [comment]
@lucaspimentel Thanks the review. 
I updated the PR with your comments.

**Comment by clamoriniere** [comment]
@lucaspimentel any plan for merging it?

**Comment by lucaspimentel** [comment]
@clamoriniere I'm not a maintainer on this repo or have write access. I think @yannmh can do it. Sorry about the confusion.

**Comment by lemkepf** [comment]
Can we get a nuget release for this? The ConstantTags would be HUGE.

**Comment by yannmh** [comment]
@lemkepf we are on it. ETA is before the week.

**Pull request merged**